### PR TITLE
Creating New Plans

### PR DESCRIPTION
**Pull request by mordonez719**
## Description

Users can create empty workouts and meal plans. When clicking the "create new" button on the homepage, a pop-up is displayed. The user is asked for a title and description of the new workout/meal plan. On submit, the plan is added to the database and is linked with the user. On the homepage, the titles of meal plans and workouts linked to the currently signed in user are displayed. Users can click each workout/meal plan card to see more information in a pop-up.

Later, users will be able to add exercises and meals from the search pages into their plans. The detailed view for each workout and meal plan will then include the individual exercises and meals pertaining to the plan.

## Milestones
- Users can create empty meal plans and view them on the homepage
- Users can create empty workout plans and view them on the homepage

## Test Plan
Screenshot 2024-07-05 at 5 29 25 PM
Screenshot 2024-07-05 at 5 31 00 PM
Screenshot 2024-07-05 at 5 31 54 PM
Screenshot 2024-07-05 at 5 32 20 PM

### CODE REVIEW

File: front-end/src/pages/login/AuthProvider.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    const authContext = useContext(AuthContext);
+
+    if (!authContext) {
+        throw new Error('useAuth must be used within a AuthProvider');
+    }
+
+    return authContext;
+}
+
+const AuthProvider = ({children}) => {
```
Comment (jessiegomes) [inline front-end/src/pages/login/AuthProvider.jsx:15]: Where and what is this used for? Some comments in this file would help.

**Review by jessiegomes — APPROVED** [review]
Overall, looks great! Left a few comments.

File: back-end/prisma/migrations/20240705203226_user_to_plans/migration.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "username" TEXT NOT NULL,
+
+    CONSTRAINT "Plan_pkey" PRIMARY KEY ("id")
+);
+
+-- AddForeignKey
+ALTER TABLE "Workout" ADD CONSTRAINT "Workout_username_fkey" FOREIGN KEY ("username") REFERENCES "User"("user") ON DELETE RESTRICT ON UPDATE CASCADE;
+
+-- AddForeignKey
+ALTER TABLE "Plan" ADD CONSTRAINT "Plan_username_fkey" FOREIGN KEY ("username") REFERENCES "User"("user") ON DELETE RESTRICT ON UPDATE CASCADE;
```
[the remark this answers is not in the document]
Comment (mordonez719) [reply back-end/prisma/migrations/20240705203226_user_to_plans/migration.sql:25]: Line 22 links the "username" field in the Workout model to the "user" field in a User model. It ensures each workout is associated with a valid user and the two fields across the models are the same. Line 25 does the same for the meal plan model.

image

File: front-end/src/pages/login/AuthProvider.jsx (hunk 1 above)
Comment (mordonez719) [reply front-end/src/pages/login/AuthProvider.jsx:15]: I was testing a potential solution to the cookies issue but the code did not work in the way I was looking for. The file will be deleted in the next pr.

**Pull request merged**